### PR TITLE
K8SPSMDB-854: Backup gets stuck after cluster expose option toggle

### PR DESCRIPTION
**Pull request by inelpandzic**
[K8SPSMDB-854](https://jira.percona.com/browse/K8SPSMDB-854) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

**CHANGE DESCRIPTION**
---
**Problem:**
When the cluster expose option gets changed, the backup gets stuck with `E [agentCheckup] get replSetGetStatus: not found` error in the backup-agent container.

**Cause:**
When replset `expose.enabled` gets toggled, mongo host in the config gets changed (from FQDN to IP if we expose the cluster) and due to PBM's lack of reading fresh node config, when performing status it compares the old value with the new.
For details check https://perconacorp.slack.com/archives/GDW1BCK6H/p1676444746488449

**Solution:**
Perform rolling restart after expose options is toggled by setting `REPLSET_EXPOSED` and `REPLSET_EXPOSED_TYPE` env variables in backup-agent containers.

**CHECKLIST**
---
**Jira**
- [x] Is the Jira ticket created and referenced properly?
- [x] Does the Jira ticket have the proper statuses for documentation (`Needs Doc`) and QA (`Needs QA`)?
- [x] Does the Jira ticket link to the proper milestone (Fix Version field)?

**Tests**
- [ ] Are E2E tests passing?
- [ ] Are unit tests passing?
- [ ] Are linting tests passing?
- [ ] Is an E2E test/test case added for the new feature/change?
- [ ] Are unit tests added where appropriate?
- [ ] Are OpenShift compare files changed for E2E tests (`compare/*-oc.yml`)?

**Config/Logging/Testability**
- [ ] Are all needed new/changed options added to default YAML files?
- [ ] Are the manifests (crd/bundle) regenerated if needed?
- [ ] Did we add proper logging messages for operator actions?
- [ ] Did we ensure compatibility with the previous version or cluster upgrade process?
- [ ] Does the change support oldest and newest supported MongoDB version?
- [ ] Does the change support oldest and newest supported Kubernetes version?

### CODE REVIEW

File: pkg/psmdb/container.go [hunk 1]
```diff
@@ -113,6 +113,14 @@ func container(ctx context.Context, cr *api.PerconaServerMongoDB, replset *api.R
...
 				Value: replset.Name,
 			},
+			{
+				Name:  "REPLSET_EXPOSED",
+				Value: strconv.FormatBool(replset.Expose.Enabled),
+			},
+			{
+				Name:  "REPLSET_EXPOSED_TYPE",
+				Value: string(replset.Expose.ExposeType),
+			},
```
Comment (egegunes) [inline pkg/psmdb/container.go:123]: do we want to restart pods if backups are not enabled? this container is `mongod`, we should put these into `backup-agent` container.

**Review by egegunes — CHANGES_REQUESTED** [review]

File: pkg/psmdb/container.go (hunk 1 above)
Comment (inelpandzic) [reply pkg/psmdb/container.go:123]: Yes, I just realized that my self.

Comment (inelpandzic) [reply pkg/psmdb/container.go]: Done: [f91c1a9](https://github.com/percona/percona-server-mongodb-operator/pull/1152/commits/f91c1a9330491b8965eb3b7091c7209a105947df)

**Comment by JNKPercona** [comment]
| Test name | Status |
| ------------- | ------------- |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/upgrade-consistency.log) |
| cross-site-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/cross-site-sharded.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/storage.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/init-deploy.log) |
| one-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/one-pod.log) |
| demand-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/demand-backup.log) |
| self-healing-chaos | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/self-healing-chaos.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/limits.log) |
| ignore-labels-annotations | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/ignore-labels-annotations.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/scaling.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/scheduled-backup.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/monitoring-2-0.log) |
| operator-self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/operator-self-healing-chaos.log) |
| security-context | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/security-context.log) |
| demand-backup-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/demand-backup-sharded.log) |
| arbiter | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/arbiter.log) |
| rs-shard-migration | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/rs-shard-migration.log) |
| demand-backup-physical | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/demand-backup-physical.log) |
| service-per-pod | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/service-per-pod.log) |
| liveness | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/liveness.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/smart-update.log) |
| version-service | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/version-service.log) |
| expose-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/expose-sharded.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/users.log) |
| data-sharded | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/data-sharded.log) |
| non-voting | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/non-voting.log) |
| demand-backup-eks-credentials | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/demand-backup-eks-credentials.log) |
| data-at-rest-encryption | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/data-at-rest-encryption.log) |
| upgrade | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/upgrade.log) |
| upgrade-sharded | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-psmdb-operator/PR-1152/9e13ff52/upgrade-sharded.log) |
| We run 30 out of 33|

commit: https://github.com/percona/percona-server-mongodb-operator/pull/1152/commits/9e13ff52decc8aba4ecb1c55c4b5430610ab33a1
image: `perconalab/percona-server-mongodb-operator:PR-1152-9e13ff52`

**Comment by egegunes** [comment]
#1155 makes this obsolete

**Pull request closed without merging**